client/src/pages/Calendar.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from "react";
+import Alert from "react-bootstrap/Alert";
+
+const Calendar = () => {
+	const [bookingsData, setBookingsData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		setIsLoading(true);
+		fetch("/api/bookings?filter=2weeks")
+			.then((response) => {
+				if (!response.ok) {
+					throw Error(
+						"There was an error getting the calendar data, Please try again"
+					);
+				}
+				return response.json();
+			})
+			.then((data) => {
+				setBookingsData(data);
+				setError(null);
+			})
+			.catch((err) => {
+				setError(err.message);
+			})
+			.finally(() => {
+				setIsLoading(false);
+			});
+	}, []);
+
+	if (isLoading) {
+		return (
+			<div className="d-flex justify-content-center">
+				<div className="spinner-grow">
+					<span className="visually-hidden">Loading...</span>
+				</div>
+			</div>
+		);
+	}
+	if (error != null) {
+		return <Alert variant="danger">{error}</Alert>;
+	}
+
+	const pickUpDates = [];
+
+	bookingsData.forEach((client) => {
+		let found = pickUpDates.find((el) => el.date == client.pick_up_date);
+		if (found) {
+			found.clients.push(client);
+		} else {
+			pickUpDates.push({ date: client.pick_up_date, clients: [client] });
+		}
+	});
+
+	return (
+		<div className="container py-5">
+			<div className="p-4  bg-dark rounded-3 no-border">
+				{" "}
+				<h2 className="text-light bg-dark">Bike pick-up calendar</h2>
+			</div>
+			{pickUpDates.map((data) => {
+				return (
+					<React.Fragment key={data.date}>
+						<h2 className="text-light bg-dark">
+							{new Date(data.date).toLocaleString("en-US", {
+								weekday: "long",
+								month: "long",
+								day: "numeric",
+							})}
+						</h2>
+						<div className="table-responsive-xxl ">
+							<table className="table align-middle table-striped table-hover">
+								<caption className="visually-hidden">Clients</caption>
+								<thead className="table-dark">
+									<tr>
+										<th scope="col">Time</th>
+										<th scope="col">Name</th>
+										<th scope="col">Amount of Bikes</th>
+									</tr>
+								</thead>
+								<tbody>
+									{data.clients.map((client) => {
+										return (
+											<tr key={client.client_id}>
+												<td>
+													{new Date(data.date).toLocaleTimeString([], { hour: "2-digit", minute: "2-digit", hour12: true })}
```

Comment:
```suggestion
													{new Date(data.date).toLocaleTimeString([], { hour: "2-digit", minute: "2-digit", hour12: true })}
													{new Date(client.pick_up_date).toLocaleTimeString("en-GB", { hour: "numeric", minute: "2-digit", hour12: true })}
```
- We want to show the pick up date of the client here
- Let's use the British locale
- In 12 hour format, we usually see something like 1:30pm instead of 01:30pm so let's do `hour: numeric`

Reply:
Ok Coung I got it. thank you so much.

Reply:
@aalassv We need to change `data.date` to `client.pick_up_date` because we want to show the time that the client is picking it up